guides/create_an_app/3-persistence.md

```diff
@@ -1,5 +1,103 @@
 # 3. Persistence
 
-Lorem ipsum dolor sit amet, consectetur adipiscing elit. Proin nibh augue, suscipit a, scelerisque sed, lacinia in, mi. Cras vel lorem. Etiam pellentesque aliquet tellus. Phasellus pharetra nulla ac diam. Quisque semper justo at risus. Donec venenatis, turpis vel hendrerit interdum, dui ligula ultricies purus, sed posuere libero dui id orci. Nam congue, pede vitae dapibus aliquet, elit magna vulputate arcu, vel tempus metus leo non est. Etiam sit amet lectus quis est congue mollis. Phasellus congue lacus eget neque. Phasellus ornare, ante vitae consectetuer consequat, purus sapien ultricies dolor, et mollis pede metus eget nisi. Praesent sodales velit quis augue. Cras suscipit, urna at aliquam rhoncus, urna quam viverra nisi, in interdum massa nibh nec erat.
+So now we have a basic chat app but once our session is closed all of our messages are lost. If we were able to create, read, update and delete from a database we would be able to maintain a history of messages or have persistence across our application.
+
+## Creating our Database
+
+First, we’ll create a migration and some code to handle persistence. Migrations are pieces of code that create or change database columns. In Phoenix, we can invoke `mix phx.gen.context` from the terminal to generate a migration:
+
+```
+$ mix ecto.gen.migration CreateMessages
+```
+
+Add to the change function, following code:
+```
+    create table(:messages) do
+      add :name, :string
+      add :message, :string
+
+      timestamps()
+    end
+```
+
+`:messages` - name for table in database.
+
+`:name`, `:message` - columns in the database table.
+
+`:string` - field type. More on DB types you can find: [Ecto types](https://hexdocs.pm/ecto/Ecto.Schema.html#module-primitive-types)
+
+Then run the migration:
+
+```
+$ mix ecto.migrate
+```
+
+**Coach:** Explain, what is changeset? Why do we validate fields?
+
+We need to define a module, that will map the database fields into an Elixir struct. Create a file, called: `message.ex` in the directory: `lib/your_app_name/conversations`.
+
+```Elixir
+defmodule WorkshopsApp.Conversations.Message do
+  use Ecto.Schema
+  import Ecto.Changeset
+
+  schema "messages" do
+    field :message, :string
+    field :name, :string
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(message, attrs) do
+    message
+    |> cast(attrs, [:name, :message])
+    |> validate_required([:name, :message])
+  end
+end
+```
+
+Let's create a module, where we will create a message. Create a file, called: `conversations.ex` in the directory: `lib/your_app_name/`.
```

Comment:
`lib/your_app_name/` => `lib/workshops_app/` to have the same paths in guides and in example app